### PR TITLE
Improved logging and Replica set compatibility

### PR DESCRIPTION
**Pull request by ssalaues**

### CODE REVIEW

File: lib/metadata/mongoclient/backend.js [hunk 1]
```diff
@@ -77,6 +84,7 @@ class MongoClientInterface {
         // initialize this backend
         MongoClient.connect(mongoUrl, (err, client) => {
             if (err) {
+                this.logger.error('error connecting to mongo server', err);
```
Comment (bennettbuchanan) [inline lib/metadata/mongoclient/backend.js]: For werelogs logging, the pattern is to have the second argument as an object containing additional information. For example, `...mongo server', { error: err });`. I guess I would assume we should maintain that for these changes as well?

**Comment by bennettbuchanan** [comment]
@ironman-machine r+

**Comment by ssalaues** [comment]
Didn't realize I had some lifting issues. Fixed the issues and should be okay now.

**Comment by bennettbuchanan** [comment]
@ironman-machine r+

**Comment by bennettbuchanan** [comment]
@ironman-machine r+

**Comment by bennettbuchanan** [comment]
@ironman-machine try

**Comment by ssalaues** [comment]
Failing the awssdk tests. Will make the necessary changes.

File: lib/metadata/mongoclient/backend.js [hunk 2]
```diff
@@ -334,7 +358,11 @@ class MongoClientInterface {
             value: objVal,
         }, {
             upsert: true,
-        }, () => cb());
+        }, err => {
+            log.error('putObjectNoVer: error putting obect with no versioning',
```
Comment (JianqinWang) [inline lib/metadata/mongoclient/backend.js]: This line is causing error ""Cannot read property 'message' of null", needs an if (err) statement

**Comment by vrancurel** [comment]
@ironman-machine r+

**Comment by LaurenSpiegel** [comment]
@ironman-machine r+

**Comment by LaurenSpiegel** [comment]
@ironman-machine r+

**Comment by LaurenSpiegel** [comment]
@ironman-machine try

**Pull request merged**